README.md

```diff
@@ -65,36 +69,41 @@ For all other failures it returns 1.
 ## Modules
 
 [Actions][3] are grouped in modules, by which they can be loaded and configured
-within pxp-agent. An example of module is given by the [Puppet module][4].
+within pxp-agent. An example of module is given by the [Puppet module][4];
+a trivial one is the [reverse module][5] that is used for testing.
 
 ### Modules interface
 
-A module is a file that provides the module metadata and an interface to
-execute its actions.
+A module is a file that provides an interface to retrieve information about
+its actions (we call such information metadata - it's a set of JSON schemas) and
+to execute actions.
 
-The module metadata is a JSON object containing the following entries:
- - `configuration`: an object that specifies options that may be included in a separate configuration file (see below);
- - `actions`: a JSON array containing specifications for the `input` and `output` of each action.
+The metadata is used by pxp-agent to acquire knowledge about the module's
+actions and validate itss configuration. For each action, the metadata specifies
+the format of the input arguments and the output results. Please refer to
+[this document][12] for more details on requirements for modules.
 
-For a trivial example of module, please refer to the [reverse module][5] used
-for testing.
+To run a given action, pxp-agent invokes the module with the action
+name. The input specified in the [PXP request][6] and other parameters will be
+then passed to the module via stdin.
 
-pxp-agent will execute a module to retrieve its metadata (by passing no
-argument) and to run a given action (by invoking the action with the input
-specified in the [PXP request][6]). Normally when pxp-agent calls a module it
-must be executable. When the `interpreter` field is specified in the
-configuration file (see below), pxp-agent will use this value to execute the
-module instead.
+pxp-agent invokes modules directly, as executables. For determining the paths of
+the executables, pxp-agent will inspect the `--modules-dir` directory and
+look for:
+
+ - **POSIX**: files without any suffix;
+ - **Windows**: files with ".bat" extension.
```

Comment:
In Windows, having a script is necessary for packaging pxp-agent.
As for the suffix name, since the user that wants to add a module has to conform to an interface, I don't think it's a real limitation.